src/header.rs

```diff
@@ -1,13 +1,97 @@
 use core::marker::PhantomData;
+use core::fmt::{Debug, Formatter};
+use core::cmp::Ordering;
 
-#[derive(Clone, Copy, Debug)]
+/// Possible Types of a [`Tag`]. The names and values are taken from the example C code
+/// at the bottom of the Multiboot2 specification.
+#[repr(u32)]
+#[derive(Copy, Clone, Debug)]
+pub enum TagType {
```

Comment:
Because we rely on the elements using the exact right indices, I think it would help to be more explicit here (e.g. `End = 0` for all of them). The comments are then not needed.